File: test/fixtures/visibility.go
package visibility

const Limit, floor = 8, 1

var Counter int

type Widget struct{}

type helper struct{}

func (w Widget) Describe() string {
	return "widget"
}

func (h helper) describe() string {
	return "helper"
}

func Shared() int {
	return Limit + floor
}

func internal() int {
	return Shared()
}

var (
	Grouped = 2
	grouped = 3
)

type Alias = string

type hiddenAlias = string

File: test/fixtures/visibility.rs
pub use std::fmt;

pub const LIMIT: usize = 8;

pub(crate) struct Widget {
    pub id: u32,
    label: String,
}

pub fn shared() -> usize {
    LIMIT
}

fn internal() -> usize {
    shared()
}

pub(super) enum Kind {
    Simple,
}

mod hidden {
    pub fn tucked() -> usize {
        2
    }
}

pub mod surface {
    pub fn reachable() -> usize {
        3
    }

    pub(crate) fn shallow() -> usize {
        4
    }
}

struct Private;

impl Private {
    pub fn helper() -> usize {
        5
    }
}

impl Widget {
    pub fn tag(&self) -> u32 {
        self.id
    }
}

fn outer() -> usize {
    pub fn nested() -> usize {
        6
    }
    nested()
}
